ompi/attribute/attribute.c

```diff
@@ -507,33 +528,68 @@ ompi_attribute_keyval_destruct(ompi_attribute_keyval_t *keyval)
             free(keyval->bindings_extra_state);
         }
 
-        opal_hash_table_remove_value_uint32(keyval_hash, keyval->key);
+        opal_hash_table_remove_value_uint32(attr_subsys->keyval_hash, keyval->key);
         FREE_KEY(keyval->key);
     }
 }
 
 
-/*
- * This will initialize the main list to store key- attribute
- * items. This will be called one time, during MPI_INIT().
- */
-int ompi_attr_init(void)
+int ompi_attr_get_ref(void)
+{
+    int ret = OMPI_SUCCESS;
+
+    OPAL_THREAD_LOCK(&attribute_lock);
+
+    if (NULL == attr_subsys) {
```

Comment:
Indentations are inconsistent.

Reply:
:+1: maybe tabs vs spaces? Also, maybe avoid using goto here either by putting the contents between the lock/unlock into a seperate function or use something like ```do { } while(false);```